api/src/main/java/org/eclipse/microprofile/rest/client/ext/DefaultClientHeadersFactoryImpl.java

```diff
@@ -46,7 +46,6 @@
     private final static Logger LOG = Logger.getLogger(CLASS_NAME);
 
     private static Optional<Config> config() {
-        Config c;
```

Comment:
What were changes in this file for?

Reply:
After upgrading the dependencies, I started seeing these compiler warnings.  I think that they are unrelated to the upgrade, but the upgrade forced my IDE to recompile everything and made the warnings more obvious.  

I put the fixes to the compiler warnings in a separate commit to keep a clean history (one commit for the Jakarta upgrade, and one for the compiler warnings).  If you want, I can separate them further by putting them in separate PRs.

Reply:
I'm a numpty, didn't notice there were two commits.